storage/pebble/receipts.go

```diff
@@ -0,0 +1,184 @@
+package pebble
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/cockroachdb/pebble"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/onflow/flow-evm-gateway/models"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-evm-gateway/storage/errors"
+	"math/big"
+	"sync"
+)
+
+var _ storage.ReceiptIndexer = &Receipts{}
+
+type Receipts struct {
+	store *Storage
+	mux   sync.RWMutex
+	first uint64
+}
+
+func NewReceipts(store *Storage) *Receipts {
+	return &Receipts{
+		store: store,
+		mux:   sync.RWMutex{},
+	}
+}
+
+func (r *Receipts) Store(receipt *gethTypes.Receipt) error {
+	r.mux.Lock()
+	defer r.mux.Unlock()
+
+	// convert to storage receipt to preserve all values
+	rr := (*models.StorageReceipt)(receipt)
+	val, err := rlp.EncodeToBytes(rr)
+	if err != nil {
+		return err
+	}
+
+	// todo batch the operations
+	height := receipt.BlockNumber.Bytes()
+	if err := r.store.set(receiptTxIDToHeightKey, receipt.TxHash.Bytes(), height); err != nil {
+		return err
+	}
+
+	if err := r.store.set(receiptHeightKey, height, val); err != nil {
+		return err
+	}
+
+	return r.store.set(bloomHeightKey, height, receipt.Bloom.Bytes())
+}
+
+func (r *Receipts) GetByTransactionID(ID common.Hash) (*gethTypes.Receipt, error) {
+	r.mux.RLock()
+	defer r.mux.RUnlock()
+
+	height, err := r.store.get(receiptTxIDToHeightKey, ID.Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	return r.getByBlockHeight(height)
+}
+
+func (r *Receipts) GetByBlockHeight(height *big.Int) (*gethTypes.Receipt, error) {
+	r.mux.RLock()
+	defer r.mux.RUnlock()
+	return r.getByBlockHeight(height.Bytes())
+}
+
+func (r *Receipts) getByBlockHeight(height []byte) (*gethTypes.Receipt, error) {
+	val, err := r.store.get(receiptHeightKey, height)
+	if err != nil {
+		return nil, err
+	}
+
+	var rcp models.StorageReceipt
+	err = rlp.DecodeBytes(val, &rcp)
+	if err != nil {
+		return nil, err
+	}
+
+	// dynamically populate the values since they are not stored to save space
+	for i, l := range rcp.Logs {
+		l.BlockHash = rcp.BlockHash
+		l.BlockNumber = rcp.BlockNumber.Uint64()
+		l.TxHash = rcp.TxHash
+		l.TxIndex = rcp.TransactionIndex
+		l.Index = uint(i)
```

Comment:
I think there's also the `revoked` field, but in our case this should always be `false` as we do not have reorgs.